features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/GraphQLConsole/GraphQLAuthentication.jsx

```diff
@@ -0,0 +1,392 @@
+                                }
+                                SetKeys(appKeys);
+                            });
+                    }
+                    setSubscriptions(subs);
+                }
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.error(error);
+                }
+                const { status } = error;
+                if (status === 404) {
+                    setFound(true);
+                }
+            });
+    }, []);
+
+
+    /**
+     *
+     * @param {React.SyntheticEvent} event
+     */
+    const handleChanges = (event) => {
+        if (event.target.name === 'selectedApplication') {
+            const promiseApp = Application.get(event.target.value);
+            let accessToken;
+            let keyType;
+
+            setSelectedApplication(event.target.value);
+
+            if (subscriptions != null && subscriptions.find((sub) => sub.applicationId
+                === selectedApplication).status === 'PROD_ONLY_BLOCKED') {
+                const sKType1 = 'SANDBOX';
+                setSelectedKeyType(sKType1);
+                keyType = 'SANDBOX';
+            } else {
+                keyType = selectedKeyType;
+            }
+
+            promiseApp
+                .then((application) => {
+                    return application.getKeys();
+                })
+                .then((appKeys) => {
+                    if (appKeys.get(keyType)) {
+                        ({ accessToken } = appKeys.get(keyType).token);
+                    }
+                    setAccessTocken(accessToken);
+                    SetKeys(appKeys);
+                });
+        } else {
+            let accessToken;
+            setSelectedKeyType(event.target.value);
+
+            if (keys.get(event.target.value)) {
+                ({ accessToken } = keys.get(event.target.value).token);
+            }
+            setAccessTocken(accessToken);
+        }
+    };
+
+
+    /**
+     * Set the environment and URLs for selected environment name
+     * @param {*} event
+     */
+    const handleEnvironemtChange = (event) => {
+        setSelectedEnvironment(event.target.value);
+        const urls = environmentObject.find((elm) => event.target.value === elm.environmentName).URLs;
+        setURLss(urls);
+    };
+
+
+    /**
+     * Handle onClick of shown access token
+     */
+    const handleClickShowToken = () => {
+        setShowToken(!showToken);
+    };
+
+
+    /**
+     * Load the access token for selected key type
+     * @param {*} event
+     */
+    const handleaccessTockenChanges = (event) => {
+        setAccessTocken(event.target.value);
+    };
+
+    /**
+     * Set the security scheme type for selected security scheme type
+     * @param {*} event
+     */
+    const handlesecuritySchemeType = (event) => {
+        setSecuritySchemeType(event.target.value);
+    };
+
+
+    return (
```

Comment:
render function is too lengthy and harder to read IMO. suggestion: break this into smaller functions.